tool/tsh/common/tsh.go

```diff
@@ -688,13 +689,57 @@ func initLogger(cf *CLIConf) {
 	}
 }
 
+// transformArgs performs OpenSSH transformation on the passed in CLI args.
+func transformArgs(args []string) ([]string, error) {
+	var cmdline []string
+
+	// If the executable name is "ssh" or "scp" transform args to "tsh ssh ..."
+	// or "tsh scp ...".
+	switch filepath.Base(os.Args[0]) {
+	case "ssh":
+		cmdline = append([]string{"ssh"}, args...)
+	case "scp":
+		cmdline = append([]string{"scp"}, args...)
+	default:
+		cmdline = args
+	}
+
+	// The following flags are OpenSSH compatibility flags. They are used for
+	// people that alias "ssh" to "tsh ssh." The following OpenSSH flags are
+	// implemented. From "man 1 ssh":
+	//
+	// * "-V Display the version number and exit."
+	// * "-T Disable pseudo-terminal allocation."
+	if slices.Contains(cmdline, "-V") {
+		modules.GetModules().PrintVersion()
+
+		return nil, &common.ExitCodeError{
+			Code: 0,
+		}
+	}
+	i := slices.Index(cmdline, "-T")
+	if i > 0 {
+		slices.Replace(cmdline, i, i+1, []string{"--no-tty"}...)
+	}
```

Comment:
Can this whole thing be replaced with one call of `slices.ReplaceFunc`?

Reply:
I don't see `ReplaceFunc` in the `slices` package?

Reply:
You're right. Oops!